openhcl/virt_mshv_vtl/src/processor/tdx/mod.rs

```diff
@@ -889,15 +889,8 @@ impl BackingPrivate for TdxBacked {
                 return true;
             }
 
-            let interruptibility: Interruptibility = this
-                .runner
-                .read_vmcs32(vtl, VmcsField::VMX_VMCS_GUEST_INTERRUPTIBILITY)
-                .into();
-
-            if (check_rflags
-                && !RFlags::from_bits(backing_vtl.private_regs.rflags).interrupt_enable())
-                || interruptibility.blocked_by_sti()
-                || interruptibility.blocked_by_movss()
+            if check_rflags
```

Comment:
I'd probably move this check down, too.

In general, you want the check that's most likely to fail to run first, even if there are some cheaper checks you could perform first that are likely to pass. I think the APIC state checks are probably going to almost always fail, whereas the other checks will almost always pass.